pkg/awsutils/awsutils.go

```diff
@@ -224,8 +225,66 @@ func prometheusRegister() {
 	}
 }
 
+//StringSet is a set of strings
+type StringSet struct {
+	sync.RWMutex
+	data 	sets.String
+}
+
+func (ss *StringSet) AWSStrings() []*string {
+	ss.RLock()
+	defer ss.RUnlock()
+	var dataSlice []*string
+	for key, _ := range ss.data {
+		dataSlice = append(dataSlice, aws.String(key))
+	}
+	return dataSlice
+}
+
+func (ss *StringSet) Set(items []string) {
+	ss.Lock()
+	defer ss.Unlock()
+	for _, item := range items {
+		ss.data.Insert(item)
```

Comment:
This needs to handle the case when `data` is `nil`. Also, it isn't actually "setting" the value of the `StringSet`s `data` field. This method should look like this instead:

```go
func (ss *StringSet) Set(items []string) {
    ss.Lock()
    defer ss.Unlock()
    ss.data = sets.NewString(items...)    
}
```